### ISSUE TITLE
Upgrade alpine to 3.6

### ISSUE THREAD

**Issue by tamalsaha** [docker-library/php#468]
Can the alpine base updated to 3.6 from 3.4? I am trying to install some additional packages and those are not available in 3.4.

**Comment by tianon** [comment]
See also https://github.com/docker-library/php/pull/357, https://github.com/docker-library/php/pull/360, and https://github.com/docker-library/golang/pull/131#issuecomment-270677556 for where a change from 3.4 to 3.5 was made previously, where it was reverted, and the discussion that led to the revert. :disappointed:

See also https://github.com/docker-library/php/pull/408 and https://github.com/docker-library/php/issues/457.

In some images, we've been adding explicit `alpine3.6` variants, but we've been hesitant to add that to PHP given that it already has an enormous number of variants as it is (and they don't build particularly fast, either).

**Comment by tamalsaha** [comment]
Thanks @tianon . I am pretty new to alpine, so I am still missing what are the issues. https://github.com/docker-library/golang/pull/131#issuecomment-270677556 says, `I'm worried this is going to break a bunch of downstreams... the switch from Alpine 3.4 to 3.5 is pretty significant.`

Are there any links/docs that might explain what are the issues.

FWIW, I am installing PHP from alpine packages in a alpine 3.6 base image. That is working fine for me.

**Comment by yosifkit** [comment]
If users are apk installing other libraries to support their php application, those could be renamed or replaced from one alpine release to the next and if we move the image it would break their builds.  We are working towards having a better way to opt in to major changes like this.

**Comment by kurlzor** [comment]
@tamalsaha If you don't want to have to maintain your own image you can also do this :
```Dockerfile
# rabbitmq-c-dev is not available in alpine 3.4 but it is in alpine 3.6
RUN apk add \
    --repository http://dl-cdn.alpinelinux.org/alpine/v3.6/main \
    --no-cache \
    rabbitmq-c-dev
```
Not sure whether it's a bad practice but it works for me.

**Comment by yosifkit** [comment]
Closing in favor of discussion to create base distro update process https://github.com/docker-library/php/issues/504.
